include/verilated_random.h

```diff
@@ -58,7 +71,25 @@ class VlRandomVar VL_NOT_FINAL {
     virtual void emitExtract(std::ostream& s, int i) const;
     virtual void emitType(std::ostream& s) const;
     virtual int totalWidth() const;
-    virtual int getLength(int dimension) const { return -1; }
+    mutable const std::map<std::string, std::shared_ptr<const ArrayInfo>>* m_arr_vars_ref
+        = nullptr;
+    virtual void
+    setArrayInfo(const std::map<std::string, std::shared_ptr<const ArrayInfo>>& arr_vars) const {
+        m_arr_vars_ref = &arr_vars;
```

Comment:
Is there an alternative to avoid storing the calling function's passed pointer?  Future spot for leaks/bugs.